### PR TITLE
Deploy local storage provisioner to public repo

### PR DESCRIPTION
**Pull request by msau42**
* Updated the Makefile to push to the public repo in quay.io
* Updated the TODO and user guide
* Fix example pvc spec

### CODE REVIEW

**Comment by msau42** [comment]
/assign @wongma7 

cc @ddysher

File: local-volume/provisioner/Makefile [hunk 1]
```diff
@@ -37,18 +28,26 @@ else
 endif
```
Comment (wongma7) [inline local-volume/provisioner/Makefile]: we can remove this block, it won't work anymore with every provisioner getting its own set of tags. Here's what I do now: https://github.com/kubernetes-incubator/external-storage/blob/master/gluster/block/Makefile#L18

File: local-volume/provisioner/Makefile [hunk 2]
```diff
@@ -37,18 +28,26 @@ else
...
+.PHONY: all build
+
+container: build quick-container
+.PHONY: container
 
-container: build
+quick-container:
 	cp local-volume-provisioner deployment/docker
-	docker build --pull -t $(IMAGE):$(VERSION) deployment/docker
+	docker build -t $(IMAGE):$(VERSION) deployment/docker
```
Comment (wongma7) [inline local-volume/provisioner/Makefile]: can add a 'docker tag' to ensure :latest is always up-to-date regardless of $(VERSION) https://github.com/kubernetes-incubator/external-storage/blob/master/gluster/block/Makefile#L46

(copied this idea from service-catalog : ))

File: local-volume/provisioner/Makefile [hunk 3]
```diff
@@ -37,18 +28,26 @@ else
...
+.PHONY: quick-container
 
 push: container
-	gcloud docker -- push $(IMAGE):$(VERSION)
+	docker push $(IMAGE):$(VERSION)
```
Comment (wongma7) [inline local-volume/provisioner/Makefile]: ditto, push $(IMAGE):latest too every time there's new version

**Comment by msau42** [comment]
Changed Makefile to look more more like gluster's.

**Pull request merged**